trade-finance-manager-ui/component-tests/utilisation-reports/utilisation-report-reconciliation-for-report.component-test.ts

```diff
@@ -179,6 +180,24 @@ describe(page, () => {
       wrapper.expectElement(`${premiumPaymentsTabSelector} input[data-cy="generate-keying-data-button"]`).notToExist();
     });
 
+    it('should not render the create record correction request button when the fee record correction feature flag is not enabled', () => {
+      const wrapper = getWrapper({
+        ...params,
+        isFeeRecordCorrectionFeatureFlagEnabled: false,
+      });
+
+      wrapper.expectElement('[data-cy="create-record-correction-request-button"]').notToExist();
+    });
+
+    it('should render the create record correction request button when the fee record correction feature flag is enabled', () => {
+      const wrapper = getWrapper({
+        ...params,
+        isFeeRecordCorrectionFeatureFlagEnabled: true,
+      });
+
+      wrapper.expectElement('[data-cy="create-record-correction-request-button"]').toExist();
+    });
```

Comment:
Can we split these into describes please? i.e

```ts
describe('when X feature flag is enabled', () => {
  it('should ...
});

describe('when X feature flag is disabled', () => {
  it('should ...
});
```

Reply:
I'd be against doing this because I think it just splits the test in half and makes it less readable especially as there is only one test per describe in your suggestion so there is also no added benefit to having a commonised test up. It's also not the pattern for the other tests in this file.

Reply:
Other instances have this approach, would like to be consitent. I think it's more readable and also helps if other conditions are added for the feature flag.

<img width="374" alt="Screenshot 2024-11-06 at 13 54 31" src="https://github.com/user-attachments/assets/7db11e86-c5be-4c5a-ab1f-4f562fb6a4bb">

Reply:
Added.

Reply:
I do think this might be good to chat about in dev friday because I personally find it much harder to work with tests with lots of nested describe blocks - I think there is an interesting trade off around maintainability and readability when it comes to describes. The worst times I have had writing tests have all been making large changes to units of code where the tests have been in nested describe blocks.